src/main/java/slimeknights/tconstruct/TConstruct.java

```diff
@@ -142,4 +170,37 @@ public Container createMenu(int p_createMenu_1_, PlayerInventory p_createMenu_2_
       });
     event.getCommandDispatcher().register(executes);
   }
+
+  @SubscribeEvent // TODO: Remove after a while, maybe at release. Doesn't work right now?
+  public void missingItemMappings(RegistryEvent.MissingMappings<Item> event) {
```

Comment:
Is this still not working? Good chance its the wrong event bus, I think this is on the FORGE bus instead of the MOD one, as it runs on world load.

Good way to check is to add a breakpoint at the start of the function, or a LOG print, and see if it is hit.

Reply:
works. It wasn't working when I originally wrote it inside the respective
submodule I was renaming blocks/items in (eg WorldBlocks). There
was also something else being stupid but I forget what it was, but its
all good now.